include/libtorrent/torrent_info.hpp

```diff
@@ -501,7 +497,7 @@ namespace libtorrent
 		// ``metadata()`` returns a the raw info section of the torrent file. The size
 		// of the metadata is returned by ``metadata_size()``.
 		int metadata_size() const { return m_info_section_size; }
-		boost::shared_array<char> metadata() const
+		std::shared_ptr<char> metadata() const
```

Comment:
I don't think this works. `shared_ptr<>` won't support arrays until C++17. This means a pointer to a single `char`